perfmetrics/scripts/ls_metrics/listing_benchmark_test.py

```diff
@@ -0,0 +1,519 @@
+                                     '80 %ile': 23.762000000000015,
+                                     '90 %ile': 57.10600000000001,
+                                     '95 %ile': 73.77799999999999,
+                                     '98 %ile': 83.7812,
+                                     '99 %ile': 87.1156,
+                                     '100 %ile': 90.45
+                                 }
+                             },
+                             '1KB_0files_0subdir':
+                             {
+                                 'Test Desc.': 'fake_test',
+                                 'Number of samples': 5,
+                                 'Mean': 37,
+                                 'Median': 21,
+                                 'Standard Dev': 39.62953444086872,
+                                 'Quantiles':
+                                 {
+                                     '0 %ile': 6.0,
+                                     '20 %ile': 6.8,
+                                     '40 %ile': 15.400000000000002,
+                                     '60 %ile': 33.0,
+                                     '80 %ile': 60.80000000000001,
+                                     '90 %ile': 80.4,
+                                     '95 %ile': 90.19999999999999,
+                                     '98 %ile': 96.08,
+                                     '99 %ile': 98.03999999999999,
+                                     '100 %ile': 100.0
+                                 }
+                             }
+                         }
+                         )
+
+    @patch('listing_benchmark.subprocess.call', return_value=1)
+    @patch('listing_benchmark.time.time', return_value=1)
+    def test_record_time_of_operation(self, mock_time, mock_subprocess_call):
+        result_list = listing_benchmark._record_time_of_operation(
+            'ls', 'fakepath/', 5)
+        self.assertEqual(mock_subprocess_call.call_count, 5)
+        self.assertEqual(result_list, [0, 0, 0, 0, 0])
+
+    @patch('listing_benchmark.subprocess.call', return_value=1)
+    @patch('listing_benchmark.time.time')
+    def test_record_time_of_operation_different_time(self, mock_time, mock_subprocess_call):
+        mock_time.side_effect = [1, 2, 3, 5]
+        result_list = listing_benchmark._record_time_of_operation(
+            'ls', 'fakepath/', 2)
+        self.assertEqual(mock_subprocess_call.call_count, 2)
+        self.assertEqual(result_list, [1000, 2000])
+
+    @patch('listing_benchmark._record_time_of_operation')
+    def test_perform_testing_type1(self, mock_record_time_of_operation):
+        mock_record_time_of_operation.return_value = [1, 1, 1]
+        gcs_bucket_results, persistent_disk_results = listing_benchmark._perform_testing(DIRECTORY_STRUCTURE1.folders,
+                                                                                         'fake_bucket', 'fake_disk', 3, 'ls -R')
```

Comment:
this formatting looks weird. Can you reformat this?

Reply:
done.